tests/unittests/core/test_space_builder.py

```diff
@@ -0,0 +1,258 @@
+        assert dim.name == 'yolo'
+        assert dim._prior_name == 'norm'
+        assert isinstance(dim.prior, dists.norm_gen)
+
+        dim = dimbuilder.build('yolo2', 'gaussian(1, 0.5, discrete=True)')
+        assert isinstance(dim, Integer)
+        assert dim.name == 'yolo2'
+        assert dim._prior_name == 'norm'
+        assert isinstance(dim.prior, dists.norm_gen)
+
+    def test_build_normal(self, dimbuilder):
+        """Check that gaussian/normal/norm is built into reciprocal correctly."""
+        dim = dimbuilder.build('yolo', 'normal(0.001, 10)')
+        assert isinstance(dim, Real)
+        assert dim.name == 'yolo'
+        assert dim._prior_name == 'norm'
+        assert isinstance(dim.prior, dists.norm_gen)
+
+        dim = dimbuilder.build('yolo2', 'normal(1, 0.5, discrete=True)')
+        assert isinstance(dim, Integer)
+        assert dim.name == 'yolo2'
+        assert dim._prior_name == 'norm'
+        assert isinstance(dim.prior, dists.norm_gen)
+
+    def test_build_enum(self, dimbuilder):
+        """Create correctly a `Categorical` dimension."""
+        dim = dimbuilder.build('yolo', "enum('adfa', 1, 0.3, 'asaga', shape=4)")
+        assert isinstance(dim, Categorical)
+        assert dim.name == 'yolo'
+        assert dim._prior_name == 'Distribution'
+        assert isinstance(dim.prior, dists.rv_discrete)
+
+        dim = dimbuilder.build('yolo', "enum(['adfa', 1])")
+        assert isinstance(dim, Categorical)
+        assert dim.name == 'yolo'
+        assert dim._prior_name == 'Distribution'
+        assert isinstance(dim.prior, dists.rv_discrete)
+
+        dim = dimbuilder.build('yolo2', "enum({'adfa': 0.1, 3: 0.4, 5: 0.5})")
+        assert isinstance(dim, Categorical)
+        assert dim.name == 'yolo2'
+        assert dim._prior_name == 'Distribution'
+        assert isinstance(dim.prior, dists.rv_discrete)
+
+        with pytest.raises(TypeError) as exc:
+            dimbuilder.build('yolo2', "enum({'adfa': 0.1, 3: 0.4})")
+        assert 'Parameter' in str(exc.value)
+        assert 'sum' in str(exc.value.__cause__)
+
+    def test_build_fails_because_empty_args(self, dimbuilder):
+        """What happens if somebody 'forgets' stuff?"""
+        with pytest.raises(TypeError) as exc:
+            dimbuilder.build('yolo', "enum()")
+        assert 'Parameter' in str(exc.value)
+        assert 'categories' in str(exc.value)
+
+        with pytest.raises(TypeError) as exc:
+            dimbuilder.build('what', "alpha()")
+        assert 'Parameter' in str(exc.value)
+        assert 'positional' in str(exc.value.__cause__)
+
+    def test_build_fails_because_troll(self, dimbuilder):
+        """What happens if somebody does not fit regular expression expected?"""
+        with pytest.raises(TypeError) as exc:
+            dimbuilder.build('yolo', "lalalala")
+        assert 'Parameter' in str(exc.value)
+        assert 'form for prior' in str(exc.value)
+
+    def test_build_random(self, dimbuilder):
```

Comment:
The support for everything (Real, Integer, Categorical) for random() is very confusing. If random is not giving any additional features for Real and Integer I would make it only support Categorical.

Suppose we make random support only Categorical, then enum would not be necessary. Unless we drop random and keep enum only. Personnaly I'd prefer random because of the support for probability distributions. It's weird inside an enum.